$NetBSD: patch-src_CMakeLists.txt,v 1.9 2021/07/13 12:36:30 markd Exp $

Don't link with the C compiler, CMake's CXXABI detection may insert
C++ libraries into LIBS.
fix for openexr3

--- src/CMakeLists.txt.orig	2017-05-28 14:22:21.000000000 +0000
+++ src/CMakeLists.txt
@@ -300,13 +300,18 @@ if(USE_CAMERA_SUPPORT)
 endif(USE_CAMERA_SUPPORT)
 
 if(USE_OPENEXR)
-  find_package(OpenEXR)
-  if(OPENEXR_FOUND)
+  find_package(OpenEXR 3.0 CONFIG)
+  if(TARGET OpenEXR::OpenEXR)
+    set(OpenEXR_LIBRARIES OpenEXR::OpenEXR)
+  else()
+    find_package(OpenEXR)
+  endif()
+  if(OpenEXR_FOUND)
     include_directories(SYSTEM ${OpenEXR_INCLUDE_DIRS})
     list(APPEND LIBS ${OpenEXR_LIBRARIES})
     add_definitions(${OpenEXR_DEFINITIONS})
     list(APPEND SOURCES "common/imageio_exr.cc")
-  endif(OPENEXR_FOUND)
+  endif(OpenEXR_FOUND)
 endif(USE_OPENEXR)
 
 if(USE_WEBP)
@@ -430,10 +435,10 @@ if(LENSFUN_FOUND)
   add_definitions("-DHAVE_LENSFUN")
 endif(LENSFUN_FOUND)
 
-if(OPENEXR_FOUND)
+if(OpenEXR_FOUND)
   add_definitions("-DHAVE_OPENEXR")
   set(DT_SUPPORTED_EXTENSIONS ${DT_SUPPORTED_EXTENSIONS} j2c j2k jp2 jpc CACHE INTERNAL "")
-endif(OPENEXR_FOUND)
+endif(OpenEXR_FOUND)
 
 if(USE_OPENCL)
   add_definitions("-DHAVE_OPENCL")
@@ -800,7 +805,7 @@ if(APPLE)
   set_target_properties(lib_darktable PROPERTIES MACOSX_RPATH TRUE)
 endif(APPLE)
 set_target_properties(lib_darktable PROPERTIES OUTPUT_NAME darktable)
-set_target_properties(lib_darktable PROPERTIES LINKER_LANGUAGE C)
+#set_target_properties(lib_darktable PROPERTIES LINKER_LANGUAGE C)
 
 #
 # Link external libs
